unhaggle/django-mass-edit
# MI-1415: Improving MassEdit processing speed

## Description
This ticket improves functionality in `django-mass-edit` in a separate file, to mitigate an issue, where Django admin users weren't able to execute mass edit, due to server's slow response and processing.

It achieves it by checking all the changes on a single object, and then applying it to all of the object, using atomic in case of database errors like constrains.

## Testing
To test this PR there was created a separate branch, that removes user requirements from the functionality, and uses a package with the newest code from this PR (at point of writing it is dev23.
1. Go to https://test-dtn--mi-1415-xpdtm.docker2.motocommerce.ca/uh-adm/moto_dealer/
2. Select any model from the list, and enter it's object list page
3. Select at least once object on from the list
4. Select "Bulk Edit Selected" action
5. Press "Go" button next to the drop down
6. After the page loads, select and change at least one of the listed fields (Object in the same line have to me marked and changed to have an effect)
7. Go to the bottom of the page and press "Save"
8. All of the objects selected to change should have edited fields with the value given during point 6, and none of the not selected fields should be changed in any of the objects


## Squash Instance
Login to the admin account: `super/super`
https://test-dtn--mi-1415-xpdtm.docker2.motocommerce.ca/uh-adm/moto_dealer/

## JIRA
https://motocommerce.atlassian.net/browse/MI-1415

## tier3 branch
https://github.com/unhaggle/tier3/pull/6694
